### PR TITLE
[LUPEYALPHA-771] check your answers

### PR DESCRIPTION
**Pull request by alkesh**
https://dfedigital.atlassian.net.mcas.ms/browse/LUPEYALPHA-771

The check-your-answers page has a text field where the provider has to enter their name.

### CODE REVIEW

File: app/controllers/submissions_controller.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ class SubmissionsController < BasePublicController
   skip_before_action :send_unstarted_claimants_to_the_start, only: [:show]
 
   def create
-    @form = journey::ClaimSubmissionForm.new(journey_session: journey_session)
+    @form = journey::ClaimSubmissionForm.new(journey_session: journey_session, params: params)
```
Comment (asmega) [inline app/controllers/submissions_controller.rb:7]: can the check your answers page not be a `Form` like most of the other pages in the journey? that way you won't have to do this params business. I think if you think of the check your answers page as a form for `submitter_name` but with a whole load of read-only content ie the check answers bit

Comment (alkesh) [reply app/controllers/submissions_controller.rb]: 👍🏽 yes - that does work - using a callback `check_your_answers_after_form_save_success` to create and save the `ClaimSubmissionForm`

File: app/models/journeys/early_years_payment/provider/authenticated/answers_presenter.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            [
+              "Employer’s PAYE reference number",
+              answers.paye_reference,
+              "paye-reference"
+            ]
+          end
+
+          def employee_name
+            [
+              "Employee's name",
```
Comment (AbigailMcP) [inline app/models/journeys/early_years_payment/provider/authenticated/answers_presenter.rb:40]: Small one, but there's a a few straight apostrophes in this file that need to be curly ’

Comment (AbigailMcP) [inline app/models/journeys/early_years_payment/provider/authenticated/answers_presenter.rb:40]: Small one, but there's a a few straight apostrophes in this file that need to be curly ’

**Review by AbigailMcP — COMMENTED** [review]
Question for design really - if i try to submit the form without adding my name, it jumps to the top of the page and the error text is off screen. I think we need an error summary at the top of the page to provide a link to the form field in error.

**Comment by alkesh** [comment]
> Question for design really - if i try to submit the form without adding my name, it jumps to the top of the page and the error text is off screen. I think we need an error summary at the top of the page to provide a link to the form field in error.

Checked with Xanthe - yes, we should have an error banner on the check-your-answers page - and also, for consistency, the other pages should also show an error banner.

File: app/models/journeys/early_years_payment/provider/authenticated/answers_presenter.rb (hunk 2 above)
Comment (alkesh) [reply app/models/journeys/early_years_payment/provider/authenticated/answers_presenter.rb]: 👍🏽 fixed

**Review by AbigailMcP — APPROVED** [review]

**Pull request merged**